src/plugins/intel_npu/src/plugin/src/plugin.cpp

```diff
@@ -37,18 +37,31 @@ const char* NPU_PLUGIN_LIB_NAME = "openvino_intel_npu_plugin";
  * @param resultDescriptors Describes the output nodes.
  * @param inputNames The names of the inputs registered in the order given by the model.
  * @param outputNames The names of the outputs registered in the order given by the model.
+ * @param config
  */
 std::shared_ptr<ov::Model> create_dummy_model(const IONodeDescriptorMap& parameterDescriptors,
                                               const IONodeDescriptorMap& resultDescriptors,
                                               const std::vector<std::string>& inputNames,
-                                              const std::vector<std::string>& outputNames) {
+                                              const std::vector<std::string>& outputNames,
+                                              const Config& config) {
     ov::ParameterVector parameters;
     ov::NodeVector results;
 
+    const auto& batchMode = config.get<BATCH_MODE>();
+
     for (const std::string& inputName : inputNames) {
         const IONodeDescriptor& parameterDescriptor = parameterDescriptors.at(inputName);
-        std::shared_ptr<ov::op::v0::Parameter> parameter =
-            std::make_shared<ov::op::v0::Parameter>(parameterDescriptor.precision, parameterDescriptor.transposedShape);
+
+        std::shared_ptr<ov::op::v0::Parameter> parameter = [&] {
```

Comment:
Actually, this is because of a bug in drivers. This is fixed with newer versions. I changed that `if` condition to `backends->isBatchSupported()`. Maybe it makes more sense now.

Reply:
Yep, looks clearer.